src/main/java/de/tum/in/www1/artemis/repository/UserRepository.java

```diff
@@ -1048,11 +1048,16 @@ SELECT COUNT(user) > 0
             INNER JOIN Exercise exercise
             ON user.login = :login
                 AND exercise.id = :exerciseId
-            INNER JOIN exercise.course course
+            LEFT JOIN exercise.course course
+            LEFT JOIN exercise.exerciseGroup.exam.course examCourse
             WHERE (course.studentGroupName MEMBER OF user.groups)
                     OR (course.teachingAssistantGroupName MEMBER OF user.groups)
                     OR (course.editorGroupName MEMBER OF user.groups)
                     OR (course.instructorGroupName MEMBER OF user.groups)
+                    OR (examCourse.studentGroupName MEMBER OF user.groups)
+                    OR (examCourse.teachingAssistantGroupName MEMBER OF user.groups)
+                    OR (examCourse.editorGroupName MEMBER OF user.groups)
+                    OR (examCourse.instructorGroupName MEMBER OF user.groups)
                     OR (:#{T(de.tum.in.www1.artemis.domain.Authority).ADMIN_AUTHORITY} MEMBER OF user.authorities)
             """)
     boolean isAtLeastStudentInExercise(@Param("login") String login, @Param("exerciseId") long exerciseId);
```

Comment:
Wouldn't it be better (optimizing for performance) to use subqueries here? 

If I am not mistaken, we are always executing at least one left join that is not necessary here (either we only need the examCourse or the course, but we will never need bought -> for one of these cases we should be able to use expression short circuiting?)

Reply:
Another option would maybe to split this into two functions, one for exam mode, one for normal exercises. This way only one left join is executed at a time